solver/src/settlement_submission/public_mempool.rs

```diff
@@ -86,14 +85,6 @@ pub async fn submit(
     }
 }
 
-async fn transaction_count(contract: &GPv2Settlement) -> Result<U256> {
-    let defaults = contract.defaults();
-    let address = defaults.from.as_ref().unwrap().address();
-    let web3 = contract.raw_instance().web3();
-    let count = web3.eth().transaction_count(address, None).await?;
-    Ok(count)
-}
```

Comment:
Looking back at this, I am not sure what we were even doing before, why would we use the settlement contract to get the transaction count of our solver address?

Reply:
It's alright, we're using contract's default `from` address which is set to be the solver's address somewhere in the `main` function.

Reply:
Yea, but it is certainly not the case that the deployer (assuming this is the contract's from address) of the settlement contract is always going to be a solver... seems very strange that we would ever have gone this direction here.

Reply:
It's not the `deployer`, it's explicitly overridden [here][1]. I.e. every interaction with this contract is sent from the solver's address. AFAIK we don't even extract deployer's address from artifacts.

[1]: https://github.com/gnosis/gp-v2-services/blob/adc937bf4113694c53dc3980286835f481383ebe/solver/src/lib.rs#L27

Reply:
Still, whatever we had here before seems awkward and unnecessarily complicated. Thanks for making it much better.